Kostyukevich/lab4.cpp

```diff
@@ -0,0 +1,93 @@
+#include "windows.h"
+#include "TlHelp32.h"
+#include <fstream>
+#include <iostream>
+#include <string>
+using namespace std;
+
+int main() {
+
+
+	ifstream deadListFile("deadList.txt");
+	ofstream logFile("log.txt");
+
+	TCHAR processName[200][MAX_PATH];
+	string processNameStr[200];
+	TCHAR buff[MAX_PATH];
+
+	char charProcessName[MAX_PATH];
+	int size = 0;
+	
+	while (deadListFile.getline(charProcessName, MAX_PATH)) {
+		processNameStr[size] = charProcessName;
+		mbstowcs(processName[size], charProcessName, strlen(charProcessName) + 1);
+		size++;
+	}
+	
+	HANDLE hSnapshot;
+	PROCESSENTRY32 processEntry;
+	BOOL result;
+
+	cout << "tracing..." << endl << "for finish program press ctrl + c" << endl;
+	
+	while (true) {
+
+
+
+		hSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+
+		if (hSnapshot == INVALID_HANDLE_VALUE) {
+			logFile << "Can't create snapshot of existing processes" << endl;
+			break;
+		}
+		
+	
+
+		processEntry.dwSize = sizeof processEntry;
+		result = Process32First(hSnapshot, &processEntry);
+		
+			while (result)
+			{
+				for (int i = 0; i < size; i++) {
+					BOOL cmp = lstrcmpi(processEntry.szExeFile, processName[i]);
+					if (cmp)
+					{
+						GetFileTitle(processEntry.szExeFile, buff, MAX_PATH);
+						cmp = lstrcmpi(buff, processName[i]);
+					}
+					if (cmp)
+					{
+						GetShortPathName(processEntry.szExeFile, buff, MAX_PATH);
+						cmp = lstrcmpi(buff, processName[i]);
+					}
+					if (!cmp)
+					{
+
+						logFile << processNameStr[i] << "  -  Find forbidden process" << endl;
```

Comment:
You should also log all process creations, according to task description. Not only attemps to kill them, but all processes that were created.
Log should not look like 
```
iexplore.exe  -  Find forbidden process
iexplore.exe  -  process succesfully terminated
iexplore.exe  -  Find forbidden process
iexplore.exe  -  process succesfully terminated
```
It should has also info about all processes that were started. Plus, it would be good to print current time to log.